config/v1/types_infrastructure.go

```diff
@@ -691,7 +691,44 @@ type BareMetalPlatformLoadBalancer struct {
 
 // BareMetalPlatformSpec holds the desired state of the BareMetal infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type BareMetalPlatformSpec struct{}
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.apiServerInternalIPs) || has(self.apiServerInternalIPs)", message="Value is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.ingressIPs) || has(self.ingressIPs)", message="Value is required once set"
+type BareMetalPlatformSpec struct {
+	// apiServerInternalIPs are the IP addresses to contact the Kubernetes API
+	// server that can be used by components inside the cluster, like kubelets
+	// using the infrastructure rather than Kubernetes networking. These are the
+	// IPs for a self-hosted load balancer in front of the API servers.
+	// In dual stack clusters this list contains two IP addresses.
+	// In single stack clusters a single IP address is expected.
+	// When omitted, values from the status.apiServerInternalIPs will be used.
+	// Once set, the list cannot be completely removed (but its second entry can).
+	//
+	// +kubebuilder:validation:Format=ip
```

Comment:
Does this work? Can you add an integration test to prove if it works or not? 

Does this support both ipv4 and ipv6?

You mention in the godoc that in dual stack clusters we expect one ipv4 and one ipv6

Reply:
`+kubebuilder:validation:Format=ip` is a copy-pasta from the already existing PlatformStatus structs

Reply:
Format IP doesn't work, we have other ways to validate IP addresses that we can look to add, will check what others are doing here

Reply:
Refactored to use custom `[]IP` with regex validation